manrs/Cisco start config.txt
as64500#sh startup-config
Using 3618 out of 262144 bytes
!
! Last configuration change at 20:28:39 UTC Sat Nov 10 2018 by manrs
!
version 15.6
service timestamps debug datetime msec
service timestamps log datetime msec
no service password-encryption
!
hostname as64500
!
boot-start-marker
boot-end-marker
!
!
enable secret 5 $1$Uljv$EmaAz7nfiISQQaJuLoE24.
!
aaa new-model
!
!
aaa authentication login default local
aaa authentication enable default enable
aaa authorization exec default local
!
!
!
!
!
aaa session-id common
ethernet lmi ce
!
!
!
mmi polling-interval 60
no mmi auto-configure
no mmi pvc
mmi snmp-timeout 180
!
!
!
!
!
no ip icmp rate-limit unreachable
!
!
!
!
!
!
no ip domain lookup
ip cef
ipv6 unicast-routing
ipv6 cef
!
multilink bundle-name authenticated
!
!
!
!
username gns3 privilege 15 secret 5 $1$tNPZ$t.Ajdj5FRfSzahkHym3vo.
username manrs privilege 15 secret 5 $1$J.n8$U7j5KHZXoT7E8djbcvQ/10
!
redundancy
!
no cdp log mismatch duplex
!
ip tcp synwait-time 5
!
!
!
!
!
!
!
!
!
!
!
!
!
interface GigabitEthernet0/0
 no shutdown
 description To AS64510 - Transit
 ip address 192.168.255.255 255.255.255.254
 no ip redirects
 ipv6 address 2001:DB8:F000:FFFF::B/127
 no ipv6 redirects
!
interface GigabitEthernet0/1
 no shutdown
 description To AS64511 - Peer
 ip address 203.0.113.250 255.255.255.254
 no ip redirects
 ipv6 address 2001:DB8:1000:FFFD::A/127
 no ipv6 redirects
!
interface GigabitEthernet0/2
 no shutdown
 description To AS64501 - Customer
 ip address 203.0.113.252 255.255.255.254
 no ip redirects
 ipv6 address 2001:DB8:1000:FFFE::A/127
 no ipv6 redirects
!
interface GigabitEthernet0/3
 no shutdown
 description To AS64502 - Customer
 ip address 203.0.113.254 255.255.255.254
 no ip redirects
 ipv6 address 2001:DB8:1000:FFFF::A/127
 no ipv6 redirects
!
router bgp 64500
 bgp log-neighbor-changes
 no bgp default ipv4-unicast
 neighbor 2001:DB8:1000:FFFD::B remote-as 64511
 neighbor 2001:DB8:1000:FFFD::B description Peer
 neighbor 2001:DB8:1000:FFFE::B remote-as 64501
 neighbor 2001:DB8:1000:FFFE::B description Customer
 neighbor 2001:DB8:1000:FFFF::B remote-as 64502
 neighbor 2001:DB8:1000:FFFF::B description Customer
 neighbor 2001:DB8:F000:FFFF::A remote-as 64510
 neighbor 2001:DB8:F000:FFFF::A description Transit
 neighbor 192.168.255.254 remote-as 64510
 neighbor 192.168.255.254 description Transit
 neighbor 203.0.113.251 remote-as 64511
 neighbor 203.0.113.251 description Peer
 neighbor 203.0.113.253 remote-as 64501
 neighbor 203.0.113.253 description Customer
 neighbor 203.0.113.255 remote-as 64502
 neighbor 203.0.113.255 description Customer
 !
 address-family ipv4
  network 203.0.113.0
  neighbor 192.168.255.254 activate
  neighbor 203.0.113.251 activate
  neighbor 203.0.113.253 activate
  neighbor 203.0.113.255 activate
 exit-address-family
 !
 address-family ipv6
  network 2001:DB8:1000::/36
  neighbor 2001:DB8:1000:FFFD::B activate
  neighbor 2001:DB8:1000:FFFE::B activate
  neighbor 2001:DB8:1000:FFFF::B activate
  neighbor 2001:DB8:F000:FFFF::A activate
 exit-address-family
!
ip forward-protocol nd
!
!
no ip http server
no ip http secure-server
ip route 203.0.113.0 255.255.255.0 Null0
!
ipv6 route 2001:DB8:1000::/36 Null0
!
!
!
!
!
control-plane
!
banner exec ^
Welcome
^
banner incoming ^
Welcome
^
banner login ^
Log in with username 'manrs' and password 'manrs':^
!
line con 0
 exec-timeout 0 0
 privilege level 15
 logging synchronous
line aux 0
 exec-timeout 0 0
 privilege level 15
 logging synchronous
line vty 0 4
 transport input none
!
no scheduler allocate
!
end


manrs/Juniper start config.txt
version 17.4R1.16;
system {
    host-name as64500;
    root-authentication {
        encrypted-password "$6$CZ0HkQey$N8uj/Dg4UG6CpbQT7hXAlj8cvRcDuf1y5nCgSI3bFeZJLdqHfq9xmNFC3my/7ZTuR/LK8r5ra7HKxxHA9OqHt."; ## SECRET-DATA
    }
    login {
        message "--- MANRS Lab ---\nLog in with username 'manrs' and password 'manrs!':";
        user manrs {
            full-name "MANRS Student";
            uid 2000;
            class super-user;
            authentication {
                encrypted-password "$6$kuVuMyg6$9dE3cFL0nmSbUDEnVERHSab6fEYCnztmqK4lXyEwtQCb4gXvo00cZiqijehYNkbNXnGt6/8HsElyvD8AJfWLE1"; ## SECRET-DATA
            }
        }
    }
    syslog {
        user * {
            any emergency;
        }
        file messages {
            any notice;
            authorization info;
        }
        file interactive-commands {
            interactive-commands any;
        }
    }
}
chassis {
    fpc 0 {
        pic 0 {
            number-of-ports 4;
        }
        lite-mode;
    }
}
interfaces {
    ge-0/0/0 {
        description "To AS64510 - Transit";
        unit 0 {
            family inet {
                address 192.168.255.255/31;
            }
            family inet6 {
                address 2001:db8:f000:ffff::b/127;
            }
        }
    }
    ge-0/0/1 {
        description "To AS64511 - Peer";
        unit 0 {
            family inet {
                address 203.0.113.250/31;
            }
            family inet6 {
                address 2001:db8:1000:fffd::a/127;
            }
        }
    }
    ge-0/0/2 {
        description "To AS64501 - Customer";
        unit 0 {
            family inet {
                address 203.0.113.252/31;
            }
            family inet6 {
                address 2001:db8:1000:fffe::a/127;
            }
        }
    }
    ge-0/0/3 {
        description "To AS64502 - Customer";
        unit 0 {
            family inet {
                address 203.0.113.254/31;
            }
            family inet6 {
                address 2001:db8:1000:ffff::a/127;
            }
        }
    }
    fxp0 {
        unit 0;
    }
}
routing-options {
    rib inet.0 {
        static {
            route 203.0.113.0/24 discard;
        }
    }
    rib inet6.0 {
        static {
            route 2001:db8:1000::/36 discard;
        }
    }
    autonomous-system 64500;
}
protocols {
    bgp {
        export accounce;
        group Transit {
            peer-as 64510;
            neighbor 192.168.255.254;
            neighbor 2001:db8:f000:ffff::a;
        }
        group Peer {
            peer-as 64511;
            neighbor 203.0.113.251;
            neighbor 2001:db8:1000:fffd::b;
        }
        group Customer1 {
            peer-as 64501;
            neighbor 203.0.113.253;
            neighbor 2001:db8:1000:fffe::b;
        }
        group Customer2 {
            peer-as 64502;
            neighbor 203.0.113.255;
            neighbor 2001:db8:1000:ffff::b;
        }
    }
}
policy-options {
    policy-statement accounce {
        then accept;
    }
}


manrs/config.py
from collections.__init__ import OrderedDict
from ipaddress import IPv4Address, IPv4Interface, IPv4Network, IPv6Address, IPv6Interface, IPv6Network

version = '0.6'

networks = OrderedDict([
    (64500, {
        'role': 'student',
        'valid_prefix_v4': IPv4Network('203.0.113.0/24'),
        'valid_prefix_v6': IPv6Network('2001:db8:1000::/36'),
    }),
    (64501, {
        'role': 'customer',
        'link_address_v4': IPv4Interface('203.0.113.253/31'),
        'link_address_v6': IPv6Interface('2001:db8:1000:fffe::b/127'),
        'peer_asn': 64500,
        'valid_prefix_v4': IPv4Network('192.0.2.0/24'),
        'valid_prefix_v6': IPv6Network('2001:db8:1001::/48'),
        'announced_prefixes_v4': [
            (IPv4Network('192.0.2.0/24'), []),
            (IPv4Network('172.16.66.0/24'), []),
            (IPv4Network('192.168.255.0/24'), []),
        ],
        'announced_prefixes_v6': [
            (IPv6Network('2001:db8:1001::/48'), []),
            (IPv6Network('2001:db8:4567::/48'), []),
            (IPv6Network('2001:db8::/32'), []),
        ],
        'bogus_address_v4': IPv4Address('192.88.99.1'),
        'bogus_address_v6': IPv6Address('3ffe::1'),
        'stolen_address_v4': IPv4Address('203.0.113.3'),
        'stolen_address_v6': IPv6Address('2001:db8:3000::3'),
    }),
    (64502, {
        'role': 'customer',
        'link_address_v4': IPv4Interface('203.0.113.255/31'),
        'link_address_v6': IPv6Interface('2001:db8:1000:ffff::b/127'),
        'peer_asn': 64500,
        'valid_prefix_v4': IPv4Network('198.51.100.0/24'),
        'valid_prefix_v6': IPv6Network('2001:db8:2002::/48'),
        'announced_prefixes_v4': [
            (IPv4Network('198.51.100.0/24'), []),
            (IPv4Network('192.0.2.64/26'), []),
            (IPv4Network('192.0.2.128/25'), []),
        ],
        'announced_prefixes_v6': [
            (IPv6Network('2001:db8:2002::/48'), []),
            (IPv6Network('2001:db8:1001:1000::/52'), []),
        ],
        'bogus_address_v4': IPv4Address('192.88.99.2'),
        'bogus_address_v6': IPv6Address('3ffe::2'),
        'stolen_address_v4': IPv4Address('192.0.2.3'),
        'stolen_address_v6': IPv6Address('2001:db8:1000::3'),
    }),
    (64510, {
        'role': 'transit',
        'link_address_v4': IPv4Interface('192.168.255.254/31'),
        'link_address_v6': IPv6Interface('2001:db8:f000:ffff::a/127'),
        'peer_asn': 64500,
        'valid_prefix_v4': IPv4Network('10.0.0.0/8'),
        'valid_prefix_v6': IPv6Network('2001:db8::/36'),
        'announced_prefixes_v4': [
            (IPv4Network('10.0.0.0/8'), [65001, 65000, 65000]),
            (IPv4Network('172.16.0.0/12'), [65001, 65002]),
            (IPv4Network('192.168.0.0/16'), [65003, 65002]),
            (IPv4Network('203.0.113.64/26'), []),
        ],
        'announced_prefixes_v6': [
            (IPv6Network('2001:db8::/36'), [65001, 65000, 65000]),
            (IPv6Network('2001:db8:1200::/40'), []),
            (IPv6Network('2001:db8:4000::/36'), [65001, 65002]),
            (IPv6Network('2001:db8:5000::/36'), [65003, 65002]),
            (IPv6Network('2001:db8:6000::/36'), [65003, 65002, 65002]),
            (IPv6Network('2001:db8:7000::/36'), [65004, 65004, 65002]),
            (IPv6Network('2001:db8:8000::/36'), [65005, 65004, 65003, 65002]),
            (IPv6Network('2001:db8:9000::/36'), [65003, 65003, 65003, 65003, 65003, 65002]),
            (IPv6Network('2001:db8:a000::/36'), [65007, 65006, 65003, 65003, 65002]),
            (IPv6Network('2001:db8:b000::/36'), [65004, 65003, 65003, 65002]),
            (IPv6Network('2001:db8:c000::/36'), [65007, 65007, 65007, 65007, 65006, 65005, 65004, 65003, 65002]),
            (IPv6Network('2001:db8:d000::/36'), [65007, 65006, 65005, 65004, 65003, 65003, 65003, 65001]),
            (IPv6Network('2001:db8:e000::/36'), [65007, 65006, 65006, 65005, 65004, 65004, 65003, 65002]),
            (IPv6Network('2001:db8:f000::/36'), [65007, 65006, 65005, 65004, 65003, 65002, 65001]),
        ],
        'bogus_address_v4': IPv4Address('192.88.99.10'),
        'bogus_address_v6': IPv6Address('3ffe::10'),
        'stolen_address_v4': IPv4Address('198.51.100.3'),
        'stolen_address_v6': IPv6Address('2001:db8:1001::3'),
    }),

    (64511, {
        'role': 'peer',
        'link_address_v4': IPv4Interface('203.0.113.251/31'),
        'link_address_v6': IPv6Interface('2001:db8:1000:fffd::b/127'),
        'peer_asn': 64500,
        'valid_prefix_v4': None,
        'valid_prefix_v6': IPv6Network('2001:db8:3000::/36'),
        'announced_prefixes_v4': [
            (IPv4Network('0.0.0.0/0'), []),
            (IPv4Network('172.16.0.0/13'), []),
        ],
        'announced_prefixes_v6': [
            (IPv6Network('::/0'), []),
            (IPv6Network('2001:db8:3000::/36'), []),
            (IPv6Network('2001:db8:6000::/36'), [65003]),
            (IPv6Network('2001:db8:7000::/37'), [65004, 65002]),
        ],
        'bogus_address_v4': IPv4Address('192.88.99.11'),
        'bogus_address_v6': IPv6Address('3ffe::11'),
        'stolen_address_v4': IPv4Address('198.51.100.3'),
        'stolen_address_v6': IPv6Address('2001:db8:2002::3'),
    }),
])
